integ/scripts/bash/rfdk-integ-e2e.sh

```diff
@@ -90,11 +92,50 @@ for COMPONENT in **/cdk.json; do
     export ${COMPONENT_NAME}_START_TIME=$SECONDS
     if [[ "$COMPONENT_NAME" != _* ]]; then
         # Excecute the e2e test in the component's scripts directory
-        cd "$INTEG_ROOT/$COMPONENT_ROOT" && ../common/scripts/bash/component_e2e.sh "$COMPONENT_NAME" || cleanup_on_failure
+        cd "$INTEG_ROOT/$COMPONENT_ROOT"
+        if [ "${RUN_TESTS_IN_PARALLEL}" = true ]; then
+            ../common/scripts/bash/component_e2e.sh "$COMPONENT_NAME" || ../common/scripts/bash/component_e2e.sh "$COMPONENT_NAME" --destroy-only &
```

Comment:
looks like we need braces here:

```suggestion
            (../common/scripts/bash/component_e2e.sh "$COMPONENT_NAME" || ../common/scripts/bash/component_e2e.sh "$COMPONENT_NAME" --destroy-only) &
```

If not, the console will wait for the former command to finish and if it fails, it will run the later one in background.

Reply:
It works like I need (execute test on background and destroy when it fails). I believe `&` applied to all command.
But I get your point and added braces for visability.

Reply:
Putting the braces around actually changes the behavior. 
Without braces, the first command runs in foreground and the second one runs in background (if the first fails). 
With braces, both, the first and the second command will run in background.

I tried by running following command and the sleep command ran in foreground blocking my console for 5s:
`(sleep 5 && echo 'task finished' && exit 1) || (echo 'task failed') &`

Whereas, running following did not block my console at all:
`((sleep 5 && echo 'task finished' && exit 1) || (echo 'task failed')) &`

Reply:
Hm. nice observing. 
But if you try to run it from shell script behavior is the same.

Reply:
Looks like `z-shell` & 'shell/bash' have different behaviors, hence it was blocking for me!